File: claude/Dockerfile.swift
FROM docker/sandbox-templates:claude-code

USER root

# ── System deps ──────────────────────────────────────────
RUN apt-get update && apt-get install -y --no-install-recommends \
    build-essential gcc g++ cmake make \
    pkg-config libssl-dev libffi-dev zlib1g-dev \
    libicu-dev libcurl4-openssl-dev libxml2-dev \
    libncurses-dev libedit-dev \
    curl wget unzip htop tree strace \
    && rm -rf /var/lib/apt/lists/*

# ── Swift toolchain (official tarball) ───────────────────
ENV SWIFT_VERSION=6.0.3
RUN ARCH=$(dpkg --print-architecture) && \
    if [ "$ARCH" = "arm64" ]; then \
        SWIFT_ARCH=aarch64; \
        SWIFT_SUFFIX="-${SWIFT_ARCH}"; \
    else \
        SWIFT_ARCH=x86_64; \
        SWIFT_SUFFIX=""; \
    fi && \
    PLATFORM_DIR="ubuntu2404${SWIFT_SUFFIX}" && \
    TARBALL="swift-${SWIFT_VERSION}-RELEASE-ubuntu24.04${SWIFT_SUFFIX}.tar.gz" && \
    URL="https://download.swift.org/swift-${SWIFT_VERSION}-release/${PLATFORM_DIR}/swift-${SWIFT_VERSION}-RELEASE/${TARBALL}" && \
    curl -fsSL "$URL" -o /tmp/swift.tar.gz && \
    mkdir -p /usr/share/swift && \
    tar xzf /tmp/swift.tar.gz -C /usr/share/swift --strip-components=1 && \
    rm /tmp/swift.tar.gz

ENV PATH="/usr/share/swift/usr/bin:${PATH}"

RUN ln -s /usr/share/swift/usr/bin/swift /usr/local/bin/swift && \
    ln -s /usr/share/swift/usr/bin/swiftc /usr/local/bin/swiftc && \
    ln -s /usr/share/swift/usr/bin/sourcekit-lsp /usr/local/bin/sourcekit-lsp && \
    ln -s /usr/share/swift/usr/bin/lldb /usr/local/bin/lldb || true

# ── Swift developer tools ────────────────────────────────
RUN mkdir -p /home/agent/.local/bin && \
    chown -R agent:agent /home/agent/.local

USER agent
RUN mkdir -p /tmp/swift-format && cd /tmp/swift-format && \
    git clone --depth 1 https://github.com/swiftlang/swift-format.git . && \
    swift build -c release --product swift-format && \
    cp .build/release/swift-format /home/agent/.local/bin/ && \
    rm -rf /tmp/swift-format || true

RUN echo 'export PATH="$HOME/.local/bin:/usr/share/swift/usr/bin:$PATH"' >> /home/agent/.bashrc

File: claude/Dockerfile.go
FROM docker/sandbox-templates:claude-code

USER root

# ── System deps ──────────────────────────────────────────
RUN apt-get update && apt-get install -y --no-install-recommends \
    build-essential gcc g++ gdb valgrind cmake make \
    pkg-config libssl-dev libffi-dev zlib1g-dev \
    libpq-dev postgresql-client \
    strace ltrace binutils elfutils \
    curl wget unzip htop tree \
    && rm -rf /var/lib/apt/lists/*

# ── Go (latest stable, detect arch) ─────────────────────
ARG GO_VERSION=1.24.4
RUN ARCH="$(dpkg --print-architecture)" && \
    case "$ARCH" in \
        amd64) GOARCH=amd64 ;; \
        arm64) GOARCH=arm64 ;; \
        *) echo "Unsupported arch: $ARCH" && exit 1 ;; \
    esac && \
    curl -fsSL "https://go.dev/dl/go${GO_VERSION}.linux-${GOARCH}.tar.gz" \
    | tar -C /usr/local -xzf -
ENV PATH="/usr/local/go/bin:/root/go/bin:${PATH}"

# ── Go tools (combined into fewer layers) ───────────────
RUN GOPATH=/root/go go install golang.org/x/tools/gopls@latest && \
    GOPATH=/root/go go install github.com/go-delve/delve/cmd/dlv@latest && \
    GOPATH=/root/go go install honnef.co/go/tools/cmd/staticcheck@latest && \
    GOPATH=/root/go go install golang.org/x/tools/cmd/goimports@latest && \
    GOPATH=/root/go go install golang.org/x/vuln/cmd/govulncheck@latest && \
    GOPATH=/root/go go install github.com/google/pprof@latest \
    || true

RUN GOPATH=/root/go go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest || true
RUN GOPATH=/root/go go install github.com/cosmtrek/air@latest || true
RUN GOPATH=/root/go go install google.golang.org/protobuf/cmd/protoc-gen-go@latest || true
RUN GOPATH=/root/go go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@latest || true

# ── Copy Go tools to agent user ─────────────────────────
RUN mkdir -p /home/agent/go/bin && \
    cp -r /root/go/bin/* /home/agent/go/bin/ 2>/dev/null || true && \
    chown -R agent:agent /home/agent/go
ENV GOPATH="/home/agent/go"
ENV PATH="${GOPATH}/bin:/usr/local/go/bin:${PATH}"

# ── protoc ───────────────────────────────────────────────
RUN ARCH="$(dpkg --print-architecture)" && \
    case "$ARCH" in \
        amd64) PROTOC_ARCH=x86_64 ;; \
        arm64) PROTOC_ARCH=aarch_64 ;; \
    esac && \
    PROTOC_VERSION=25.2 && \
    curl -fsSL "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-${PROTOC_ARCH}.zip" \
    -o /tmp/protoc.zip && \
    unzip /tmp/protoc.zip -d /usr/local && \
    rm /tmp/protoc.zip

USER agent

RUN echo 'export GOPATH="$HOME/go"' >> /home/agent/.bashrc && \
    echo 'export PATH="$GOPATH/bin:/usr/local/go/bin:$PATH"' >> /home/agent/.bashrc

USER root
RUN claude update --yes 2>/dev/null || true
USER agent
